tests/virtualization/universal/prepare_guests.pm

```diff
@@ -20,6 +20,12 @@ sub run {
     my $self = shift;
     # Use serial terminal, unless defined otherwise. The unless will go away once we are certain this is stable
     $self->select_serial_terminal unless get_var('_VIRT_SERIAL_TERMINAL', 1) == 0;
+    # Clean up VMs if any. This makes it possable to reinstall VMs.
+    systemctl("restart libvirtd");
+    assert_script_run('for i in $(virsh list --name|grep sles);do virsh shutdown $i;done');
+    sleep 30;
+    assert_script_run('for i in $(virsh list --name|grep sles);do virsh destroy $i;done');
+    assert_script_run('for i in $(virsh list --name --inactive); do virsh undefine $i --remove-all-storage;done');
```

Comment:
How about the windows guest?

Reply:
@tonyyuan1  @tbaev  I guess this code block is the part that you say can reject the guest installation retry ticket? If yes, IMHO, it is not the same. 
This code block ensures that each time you rerun the prepare_guests.pm, all guests are cleaned and reinstalled freshly. This is beneficial and good when only 1~2 guests are tested at the same time, but it is not efficient enough when multiple guests are included in one test. Eg, you test 6 guests, 5 passed and 1 failed in installation. If using this code block, the 5 guests have to be recreated again.
However that guest retry ticket is that DURING the prepare_guests.pm step, if one guest fails to be installed, it retries multiple times by cleaning up this guest only -> reinstall guest only until it is successful or fail after max retries. It will be useful for above 6 guest example. So, IMHO, the guest retry ticket should be reopened again. HDYT?

Reply:
Which ticket is rejected?

Reply:
https://progress.opensuse.org/issues/102984

Reply:
> How about the windows guest?

windows guest is imported. It just removes the downloaded image.

Reply:
By the way, I have never seen any windows guest running in the test.

Reply:
> https://progress.opensuse.org/issues/102984
Below is what I understand, 
There is ~30 mins waiting time limit. The all guest install must be complete within 30 mins. The failure must be found as early as possible.  The second/third retry must be started as early as possible so that second/third installation could finish within remaining time before timeout. 

Currently, we reinstall all if any guest install fails. Teodor is exploring a better solution, https://progress.opensuse.org/issues/109328

Alice, as  your ticket exposes the issue we need to resolve. I would think that we could keep it open to track this issue. Maybe associate the two tickets. It's up to your two' decision :)

Reply:
Okay, given that neither this PR(cleanup all guest when test rerun) nor https://progress.opensuse.org/issues/109328 (fail earlier when guest install fails) actually do guest installation retry, I'd suggest to reopen https://progress.opensuse.org/issues/102984. @tbaev HDYT? Reasonable?

Reply:
> By the way, I have never seen any windows guest running in the test.

windows test is run after qam-kvm|xen-install, eg http://openqa.qam.suse.cz/tests/36488#dependencies

Reply:
I agree, I will link the two tickets together (109328 and 102984)